flask/src/main/groovy/net/corda/gradle/flask/FlaskPlugin.groovy

```diff
@@ -0,0 +1,69 @@
+package net.corda.gradle.flask
+
+import org.gradle.api.GradleException
+import org.gradle.api.Plugin
+import org.gradle.api.Project
+import org.gradle.api.file.Directory
+import org.gradle.api.plugins.JavaApplication
+import org.gradle.api.plugins.JavaPlugin
+import org.gradle.api.plugins.JavaPluginConvention
+import org.gradle.api.provider.Provider
+import org.gradle.api.tasks.Copy
+import org.gradle.api.tasks.JavaExec
+import org.gradle.api.tasks.SourceSet
+import org.gradle.jvm.tasks.Jar
+
+class FlaskPlugin implements Plugin<Project> {
+
+    private static final String GROUP = "Flask"
+
+    @Override
+    void apply(Project project) {
+        project.getPluginManager().apply(JavaPlugin.class)
+        Provider<Directory> flaskDir = project.layout.buildDirectory.dir("classes/flask-launcher")
+        Provider<Copy> extractLauncherTarProvider = project.tasks.register("extractLauncherTar", Copy) {
+            setGroup(GROUP)
+            setDescription("Extract the Flask Launcher classes to be used to build a custom launcher")
+            into(flaskDir)
+            from(project.tarTree(LauncherResource.instance))
+        }
+
+        JavaPluginConvention javaPluginConvention = project.convention.getPlugin(JavaPluginConvention.class)
+        SourceSet flaskSourceSet = javaPluginConvention.sourceSets.create("flask")
+        flaskSourceSet.compiledBy(extractLauncherTarProvider)
+        project.configurations.getByName(flaskSourceSet.compileOnlyConfigurationName).withDependencies { dependencies ->
+            def launcherDependency = project.dependencies.create(extractLauncherTarProvider.get().outputs.files)
+            dependencies.add(launcherDependency)
+        }
+
+        Provider<FlaskJarTask> flaskJarTask = project.tasks.register("flaskJar", FlaskJarTask.class) {
+            setGroup(GROUP)
+            setDescription("Package the current project code in an executable jar file")
+            archiveBaseName = "${project.name}-flask"
+            inputs.files(flaskSourceSet.output)
+            from {
+                flaskSourceSet.runtimeClasspath.collect {
+                    if(it.exists()) {
+                        it.isDirectory() ? it : project.zipTree(it)
+                    } else {
+                        null
+                    }
+                }
+            }
+            includeLibraries(project.tasks.named("jar", Jar).map {it.outputs })
+            includeLibraries(project.configurations.named("runtimeClasspath"))
```

Comment:
To be really, _really_ picky :smiling_imp: 
- "jar" -> `JavaPlugin.JAR_TASK_NAME`
- "runtimeClasspath" -> `JavaPlugin.RUNTIME_CLASSPATH_CONFIGURATION_NAME`

Reply:
👍🏿